packages/strapi-parts/src/TextInput/TextInput.stories.mdx

```diff
@@ -11,11 +12,32 @@ import { Tooltip } from '../Tooltip';
 
 # TextInput
 
-This is the doc of the `TextInput` component
+TextInputs are inputs that allow users to add or edit text. They support several text formats including numbers.
 
-## TextInput
+**Best practices**
 
-Description...
+- Icons are always on the right side within the input (writing area) itself, never on the left side.
+- TextButton component should always be above the input. 
+- Labels should be clear and concise. A few words only and without punctuation.
+- Use description line as a help to give more details about the required action.
+- Within a form, unless users need a lot of space to write their content, the input should take 1 column space instead of two.
+- Make sure to use the placeholder feature whenever clarification are helpful on how to use the input.
+- Make sure to use to define the input as required whenever it is absolutely mandatory.
+- If a single line is not enough for the amount of text required, use TextArea component.
+- The error message will replace the description line. The two of them cannot be on top of each other.
+- The error message can be customized, make sure to write actionable information within it to help resolve an issue.
+
+
+
+## Usage
+
+TextInputs are mostly found in forms and modals. They give users the possibility to add or edit text. 
```

Comment:
They will always be in a form. They can't be outside. Even in a modal, it's going to be a form